crates/eips/src/eip7594/sidecar.rs

```diff
@@ -0,0 +1,289 @@
+use crate::{
+    eip4844::{
+        deserialize_blobs, kzg_to_versioned_hash, Blob, BlobTransactionSidecarEip4844,
+        BlobTransactionValidationError, Bytes48,
+    },
+    eip7594::{CELLS_PER_EXT_BLOB, EIP_7594_WRAPPER_VERSION},
+};
+use alloy_primitives::B256;
+use alloy_rlp::{Buf, BufMut, Decodable, Encodable, Header};
+
+/// This represents a set of blobs, and its corresponding commitments and proofs.
+/// Proof type depends on the sidecar variant.
+///
+/// This type encodes and decodes the fields without an rlp header.
+#[derive(Clone, PartialEq, Eq, Hash, Debug)]
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+pub enum BlobTransactionSidecar {
```

Comment:
I suggest we rename this `BlobTransactionSidecarVariant` following what we did for 4844 tx type